### DISCUSSION TITLE
How to migrate from `extends: [plugin:foo]` to the new config system

### DISCUSSION THREAD

**Discussion by jjangga0214** [eslint/eslint#16291] [Q&A] [answered]
Hi!
I've just found the [new config system](https://eslint.org/docs/latest/user-guide/configuring/configuration-files-new), and am trying to use it.

My current eslintrc (old config system) is like the snippet below

**eslintrc.js**

```js
// Others are omitted for brevity. 
plugins: ['unicorn'],
extends: [
    'eslint:recommended',
    'plugin:unicorn/recommended',
]
```

**eslint.config.js**

Would this be okay?

```js
import unicorn from 'eslint-plugin-unicorn'

export default [
  'eslint:recommended',
  {
    files: ['**/*.js'],
    plugins: {
      unicorn
    },
  },
]
```

**eslint.config.js**

Or should it be like this?

```js
import unicorn from 'eslint-plugin-unicorn'

export default [
  'eslint:recommended',
  unicorn.configs.recommended,
  {
    files: ['**/*.js'],
    plugins: {
      unicorn
    },
  },
]
```

Thanks :)

**Comment by jjangga0214** [comment, marked as answer]
It should be like this.

```js
import unicorn from 'eslint-plugin-unicorn'

export default [
  'eslint:recommended',
  {
    files: ['**/*.js'],
    plugins: {
      unicorn
    },
    rules: unicorn.configs.recommended.rules
  },
]
```

However, that's because the plugin(unicorn, in this case) has not transitioned to the new config system.
Once the plugin support the new system, then it should be like this.
(Note that this time sharable config is imported, not plugin)

```js
import unicorn from 'eslint-plugin-unicorn/recommended'

export default [
  'eslint:recommended',
  unicorn,
]
```

If you want to override some properties(e.g. files), you can do so by

```js
import unicorn from 'eslint-plugin-unicorn/recommended'

export default [
  'eslint:recommended',
  {
  ...unicorn,
  files: ['the-pattern-I-want'],
  }
]
```

or


```js
import unicorn from 'eslint-plugin-unicorn/recommended'

export default [
  'eslint:recommended',
  unicorn,
  {
  files: ['**/*{js,ts,jsx,tsx}'],
  //  some override here 
  }
]
```

**Reply by AndreaPontrandolfo** [reply]
This really should be added in the docs:https://eslint.org/docs/latest/user-guide/configuring/configuration-files-new

**Reply by burtek** [reply]
And also what should be added is how to create sharable config using new system.

**Reply by Rec0iL99** [reply]
@burtek track #13481. Phase 4 includes switching the docs from legacy config to flat config for shareable configs.
